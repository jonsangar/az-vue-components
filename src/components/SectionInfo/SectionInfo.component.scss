.sectionInfo{
  display: flex;
  justify-content: space-between;
  color: color(gray3);

  &__info{
    color: inherit;
  }

  &__info--brand{
    color: color(brand);
  }

  &__info--success{
    color: color(success);
  }

  &__info--warning{
    color: color(warning);
  }

  &__info--danger{
    color: color(danger);
  }


}
