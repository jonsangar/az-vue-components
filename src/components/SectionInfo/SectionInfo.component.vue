<template lang="pug">
.sectionInfo
  .sectionInfo__title {{ title }}
  .sectionInfo__info(
    :class="getClass") {{ info }}
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: ''
    },

    info: {
      type: String,
      default: ''
    },

    mode: {
      type: String,
      default: ''
    }

  },

  computed: {
    getClass() {
      return this.mode ? `sectionInfo__info--${this.mode}` : ''
    }
  }

}
</script>

<style lang="scss" src="./SectionInfo.component.scss" scoped></style>
