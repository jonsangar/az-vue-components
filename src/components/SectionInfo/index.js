import SectionInfo from './SectionInfo.component.vue';

export default SectionInfo;
