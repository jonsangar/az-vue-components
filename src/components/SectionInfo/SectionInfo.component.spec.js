import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerÃ­as de estilos e iconos
import SvgIcon from 'vue-svgicon'
import SectionInfo from './SectionInfo.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('SectionInfo.vue', () => {
  const build = () => {
    const wrapper = shallowMount(SectionInfo, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente SectionInfo con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que se actualizan las props cuando se asignan las props', async () => {
    const { wrapper } = build()
    wrapper.setProps({
      title: 'MyTitle',
      info: 'Info',
      mode: 'brand',
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.title).toBe('MyTitle')
    expect(wrapper.vm.info).toBe('Info')
    expect(wrapper.vm.mode).toBe('brand')
  });

  it('Comprueba que se renderiza las props cuando se asignan las props', async () => {
    const { wrapper } = build()
    wrapper.setProps({
      title: 'MyTitle',
      info: 'Info',
      mode: 'brand',
    })
    await wrapper.vm.$nextTick()

    expect(wrapper.find('.sectionInfo__title').text()).toBe('MyTitle')
    expect(wrapper.find('.sectionInfo__info--brand').exists()).toBe(true)
    expect(wrapper.find('.sectionInfo__info--brand').text()).toBe('Info')
    expect(wrapper.html()).toMatchSnapshot()
  });

});
