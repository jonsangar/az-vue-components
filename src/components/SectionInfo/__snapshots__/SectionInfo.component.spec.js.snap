// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionInfo.vue Comprueba que se renderiza las props cuando se asignan las props 1`] = `
<div class="sectionInfo">
  <div class="sectionInfo__title">MyTitle</div>
  <div class="sectionInfo__info sectionInfo__info--brand">Info</div>
</div>
`;

exports[`SectionInfo.vue Comprueba renderizado del componente SectionInfo con snapshot 1`] = `
<div class="sectionInfo">
  <div class="sectionInfo__title"></div>
  <div class="sectionInfo__info"></div>
</div>
`;
